source/use-cases/client-side-field-level-encryption-local-key-to-kms.txt

```diff
@@ -48,6 +48,13 @@ Convert to a Remote Master Key
    Failure to decrypt all data at this stage may cause permanent and
    unrecoverable data loss.
 
+The following diagram shows how the **customer master key** and **data
+encryption keys** are created and stored when using a KMS provider for
+MongoDB CSFLE:
+
+.. image:: /figures/CSFLE_Create_Keys_KMS.png
```

Comment:
Added diagram to the KMS guide.

Reply:
Shouldn't the Client Application have to read the master key from AWS KMS similarly to how it does for the local key diagram?

![image](https://user-images.githubusercontent.com/15657698/70644582-4e2fd200-1c11-11ea-9a9e-93c198679431.png)
![image](https://user-images.githubusercontent.com/15657698/70644597-57b93a00-1c11-11ea-9cfa-5355ad7685ec.png)

Reply:
Yes, good catch, will update the diagram. For accuracy, the data encryption key is just passed to the AWS KMS API by libmongocrypt which returns the encrypted data encryption key.